frames/src/main/java/com/checkout/frames/utils/extensions/BillingAddressDetailsExtensions.kt

```diff
@@ -0,0 +1,107 @@
+import com.checkout.base.model.Country
+import com.checkout.frames.R
+import com.checkout.frames.component.billingaddressfields.BillingAddressInputComponentState
+import com.checkout.frames.screen.billingaddress.billingaddressdetails.models.BillingAddress
+import com.checkout.frames.screen.billingaddress.billingaddressdetails.models.BillingFormFields
+import com.checkout.tokenization.model.Phone
+
+internal fun List<BillingAddressInputComponentState>.saveBillingAddressDetails(country: Country): BillingAddress {
+    val billingAddress = BillingAddress()
+
+    val phoneInputComponentState = this.find {
+        it.addressFieldName == BillingFormFields.Phone.name
+    }
+
+    val addressLineOneInputComponentState = this.find {
+        it.addressFieldName == BillingFormFields.AddressLineOne.name
+    }
+
+    val addressLineTwoInputComponentState = this.find {
+        it.addressFieldName == BillingFormFields.AddressLineTwo.name
+    }
+
+    val stateInputComponentState = this.find {
+        it.addressFieldName == BillingFormFields.State.name
+    }
+
+    val cityInputComponentState = this.find {
+        it.addressFieldName == BillingFormFields.City.name
+    }
+
+    val zipInputComponentState = this.find {
+        it.addressFieldName == BillingFormFields.PostCode.name
+    }
+
+    billingAddress.phone = Phone(
+        number = phoneInputComponentState?.addressFieldText?.value ?: "",
+        country = country
+    )
+    billingAddress.address?.let { address ->
+        address.country = country
+        address.addressLine1 = addressLineOneInputComponentState?.addressFieldText?.value ?: ""
+        address.addressLine2 = addressLineTwoInputComponentState?.addressFieldText?.value ?: ""
+        address.state = stateInputComponentState?.addressFieldText?.value ?: ""
+        address.city = cityInputComponentState?.addressFieldText?.value ?: ""
+        address.zip = zipInputComponentState?.addressFieldText?.value ?: ""
+    }
+
+    return billingAddress
+}
+
+@StringRes
+internal fun BillingAddressInputComponentState.getErrorMessage(): Int {
+    return when (this.addressFieldName) {
+        BillingFormFields.AddressLineOne.name -> {
+            R.string.cko_billing_form_input_field_address_line_one_error
+        }
+        BillingFormFields.AddressLineTwo.name -> {
+            R.string.cko_billing_form_input_field_address_line_two_error
+        }
+        BillingFormFields.City.name -> {
+            R.string.cko_billing_form_input_field_city_error
+        }
+        BillingFormFields.State.name -> {
+            R.string.cko_billing_form_input_field_state_error
+        }
+        BillingFormFields.PostCode.name -> {
+            R.string.cko_billing_form_input_field_post_code_error
+        }
+        BillingFormFields.Phone.name -> {
+            R.string.cko_billing_form_input_field_phone_number_error
+        }
+        else -> {
+            0
+        }
+    }
+}
+
+internal fun BillingAddress.updateAddressFieldText(
```

Comment:
As far as I'm concerned, this function updates nothing just provides a text for a specific field type.

Reply:
Yes. Make sense. Updating naming convention to provideAddressFieldText.